src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -419,11 +419,11 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         else
         {
             // we'll write out one command per key mapping
-            for (auto keys{ _keyMappings.begin() }; keys != _keyMappings.end(); ++keys)
+            for (const auto& keys : _keyMappings)
             {
                 Json::Value cmdJson{ Json::ValueType::objectValue };
 
-                if (keys == _keyMappings.begin())
+                if (keys == *_keyMappings.begin())

```

Comment:
absolutely do not do this ;P

what you had before was correct -- you want to check "am I at the beginning of the list?". You do not want to check, "am I at an item that is equivalent to the item at the beginning of the list" (!!)